docs/lazy-adr/adr-004-mvp-light-client.md

```diff
@@ -0,0 +1,292 @@
+2. (aka 1.5) the DAS light client stages headers where the tendermint verification passes as valid and spins up DAS sampling rotines in the background; the staged headers are committed as valid iff all routines successfully return in time
+3. the DAS light client optimistically accepts a header as valid and trusts it if the regular tendermint verification succeeds; the DAS is run in the background (with potentially much longer timeouts as in 1.) and after the background routine returns (or errs or times out), the already trusted headers are marked as unavailable; this might require rolling back the already trusted headers
+
+We note that from an implementation point of view 1. is not only the simplest approach, but it would also work best with the currently implemented light client design.
+It is the approach that should be implemented first.
+
+The 2. approach can be seen as an optimization where the higher latency DAS can be conducted in parallel for various heights.
+This could speed up catching-up (sequentially) if the light client went offline (shorter than the weak subjectivity time window).
+
+The 3. approach is the most general of all, but it moves the responsibility to wait or to rollback headers to the caller and hence is undesirable as it offers too much flexibility.
+
+
+#### Data Structures
+
+##### LightBlock
+
+As mentioned above the LightBlock should optionally contain the DataAvailabilityHeader.
+```diff
+Index: types/light.go
+===================================================================
+diff --git a/types/light.go b/types/light.go
+--- a/types/light.go	(revision 64044aa2f2f2266d1476013595aa33bb274ba161)
++++ b/types/light.go	(date 1620481205049)
+@@ -13,6 +13,9 @@
+ type LightBlock struct {
+ 	*SignedHeader `json:"signed_header"`
+ 	ValidatorSet  *ValidatorSet `json:"validator_set"`
++
++	// DataAvailabilityHeader is only populated for DAS light clients for others it can be nil.
++	DataAvailabilityHeader *DataAvailabilityHeader `json:"data_availability_header"`
+ }
+```
+
+Alternatively, we could introduce a `DASLightBlock` that embeds a `LightBlock` and has the `DataAvailabilityHeader` as the only (non-optional) field.
+This would be more explict as it is a new type.
+Instead, adding a field to the existing `LightBlock`is backwards compatible and does not require any further code changes; the new type requires `To`- and `FromProto` functions at least.
+
+##### Provider
+
+The [`Provider`](https://github.com/tendermint/tendermint/blob/7f30bc96f014b27fbe74a546ea912740eabdda74/light/provider/provider.go#L9-L26) should be changed to additionally provide the `DataAvailabilityHeader` to enable DAS light clients.
+Implementations of the interface need to additionally to retrieve the `DataAvailabilityHeader` for the [modified LightBlock](#lightblock).
```

Comment:
```suggestion
Implementations of the interface need to additionally retrieve the `DataAvailabilityHeader` for the [modified LightBlock](#lightblock).
```

Reply:
Or they can get it from `ResultCommit.SignedHeader.Commit.BlockID.DAHeader` if #304 lands before MVP, not to additional request overhead

Reply:
> if #304 lands before MVP,

You mean #312?

Yeah, I intentionally didn't want to rely on #312. I can't think of a use-case where the DAHeader would be requested in it's own but I guess it certainly won't harm if it can be requested. We can remove the rpc endpoint / the additional method in the provider later if we wanted.

Reply:
> You mean #312?
Yes

Reply:
> I can't think of a use-case where the DAHeader would be requested in it's own but I guess it certainly won't harm if it can be requested. We can remove the rpc endpoint / the additional method in the provider later if we wanted. It does not introduce any technical debt.

Ok